numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -1489,10 +1489,14 @@ OBJECT_to_@TOTYPE@(void *input, void *output, npy_intp n,
 
     for (i = 0; i < n; i++, ip++, op += skip) {
         if (*ip == NULL) {
-            @TOTYPE@_setitem(Py_False, op, aop);
+            if (@TOTYPE@_setitem(Py_False, op, aop)) {
```

Comment:
`func(...) < 0` is the preferred idiom for these checks. I assume an error is set on failure and checked somewhere down the call chain?